File: app/src/main/java/com/satsumasoftware/pokedex/framework/nature/Nature.java
package com.satsumasoftware.pokedex.framework.nature;

import android.database.Cursor;

import com.satsumasoftware.pokedex.db.NaturesDBHelper;

public class Nature extends BaseNature {

    private int mDecreasedStatId, mIncreasedStatId, mHatesFlavorId, mLikesFlavorId;
    private String mNameJa, mNameKo, mNameFr, mNameDe, mNameEs, mNameIt;

    public Nature(int id, int decreasedStatId, int increasedStatId, int hatesFlavorId,
                  int likesFlavorId, String name, String nameJa, String nameKo, String nameFr,
                  String nameDe, String nameEs, String nameIt) {
        mId = id;
        mDecreasedStatId = decreasedStatId;
        mIncreasedStatId = increasedStatId;
        mHatesFlavorId = hatesFlavorId;
        mLikesFlavorId = likesFlavorId;

        mName = name;
        mNameJa = nameJa;
        mNameKo = nameKo;
        mNameFr = nameFr;
        mNameDe = nameDe;
        mNameEs = nameEs;
        mNameIt = nameIt;
    }

    public Nature(Cursor cursor) {
        mId = cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_ID));
        mDecreasedStatId = cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_DECREASED_STAT_ID));
        mIncreasedStatId = cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_INCREASED_STAT_ID));
        mHatesFlavorId = cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_HATES_FLAVOR_ID));
        mLikesFlavorId = cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_LIKES_FLAVOR_ID));

        mName = cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME));
        mNameJa = cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_JAPANESE));
        mNameKo = cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_KOREAN));
        mNameFr = cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_FRENCH));
        mNameDe = cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_GERMAN));
        mNameEs = cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_SPANISH));
        mNameIt = cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_ITALIAN));
    }

    public int getDecreasedStatId() {
        return mDecreasedStatId;
    }

    public int getIncreasedStatId() {
        return mIncreasedStatId;
    }

    public int getHatesFlavorId() {
        return mHatesFlavorId;
    }

    public int getLikesFlavorId() {
        return mLikesFlavorId;
    }

    public String getNameJapanese() {
        return mNameJa;
    }

    public String getNameKorean() {
        return mNameKo;
    }

    public String getNameFrench() {
        return mNameFr;
    }

    public String getNameGerman() {
        return mNameDe;
    }

    public String getNameSpanish() {
        return mNameEs;
    }

    public String getNameItalian() {
        return mNameIt;
    }

}

File: app/src/main/java/com/satsumasoftware/pokedex/framework/nature/Nature.kt
/*
 * Copyright 2016-2017 Farbod Salamat-Zadeh
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.satsumasoftware.pokedex.framework.nature

import android.database.Cursor
import com.satsumasoftware.pokedex.db.NaturesDBHelper

class Nature(val id: Int, val decreasedStatId: Int, val increasedStatId: Int,
             val hatesFlavorId: Int, val likesFlavorId: Int, val name: String,
             val japaneseName: String, val koreanName: String, val frenchName: String,
             val germanName: String, val spanishName: String, val italianName: String) {

    constructor(cursor: Cursor) : this(
            cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_ID)),
            cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_DECREASED_STAT_ID)),
            cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_INCREASED_STAT_ID)),
            cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_HATES_FLAVOR_ID)),
            cursor.getInt(cursor.getColumnIndex(NaturesDBHelper.COL_LIKES_FLAVOR_ID)),
            cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME)),
            cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_JAPANESE)),
            cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_KOREAN)),
            cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_FRENCH)),
            cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_GERMAN)),
            cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_SPANISH)),
            cursor.getString(cursor.getColumnIndex(NaturesDBHelper.COL_NAME_ITALIAN)))

}
